packages/runtime/src/next/components/head-tags.tsx

```diff
@@ -0,0 +1,66 @@
+import { ComponentPropsWithoutRef } from 'react'
+import { useIsPagesRouter } from '../../components/hooks/useIsPagesRouter'
+import Head from 'next/head'
+
+export function PageTitle({ children, ...props }: ComponentPropsWithoutRef<'title'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <title {...props}>{children}</title>
+      </Head>
+    )
+  }
+  return <title {...props}>{children}</title>
+}
+
+export function PageMeta(props: ComponentPropsWithoutRef<'meta'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <meta {...props} />
+      </Head>
+    )
+  }
+  return <meta {...props} />
+}
+
+// https://react.dev/reference/react-dom/components/link#props
+type ReactLinkProps = {
+  precedence?: 'reset' | 'low' | 'medium' | 'high'
+}
```

Comment:
If we're going to define the types, we might as well augment the React DOM module in this file.

Reply:
Decided not to do this to avoid exposing these props to _all_ `link` and `style` elements.